src/Management/src/Endpoint/ManagementPort/ManagementPortMiddleware.cs

```diff
@@ -52,18 +54,47 @@ public async Task InvokeAsync(HttpContext context)
         }
     }
 
-    private static bool IsRequestAllowed(HttpRequest request, ManagementOptions managementOptions)
+    private bool IsRequestAllowed(HttpRequest request, ManagementOptions managementOptions)
     {
-        if (int.TryParse(managementOptions.Port, CultureInfo.InvariantCulture, out int managementPort) && managementPort > 0)
+        if (!int.TryParse(managementOptions.Port, CultureInfo.InvariantCulture, out int managementPort) || managementPort <= 0)
         {
-            bool isManagementPath = request.Path.StartsWithSegments(managementOptions.Path);
-            bool isManagementScheme = managementOptions.SslEnabled ? request.Scheme == Uri.UriSchemeHttps : request.Scheme == Uri.UriSchemeHttp;
-            bool isManagementPort = request.Host.Port == managementPort;
+            return true;
+        }
+
+        bool isManagementPath = request.Path.StartsWithSegments(managementOptions.Path);
+        bool isManagementScheme = managementOptions.SslEnabled ? request.Scheme == Uri.UriSchemeHttps : request.Scheme == Uri.UriSchemeHttp;
+        bool isManagementPort = request.Host.Port == managementPort;
+
+        string? instancePorts = Environment.GetEnvironmentVariable("CF_INSTANCE_PORTS");
 
-            return isManagementPath ? isManagementScheme && isManagementPort : !isManagementScheme || !isManagementPort;
+        if (!isManagementPort && !string.IsNullOrEmpty(instancePorts))
+        {
+            isManagementPort = EvaluateCfInstancePorts(managementPort, instancePorts, request.Host.Port);
         }
 
-        return true;
+        return isManagementPath ? isManagementScheme && isManagementPort : !isManagementScheme || !isManagementPort;
+    }
+
+    private bool EvaluateCfInstancePorts(int managementPort, string instancePorts, int? requestPort)
+    {
+        var portMappings = JsonSerializer.Deserialize<List<PortMapping>>(instancePorts);
+
+        if (portMappings == null)
+        {
+            return false;
+        }
+
+        PortMapping? portMapping = portMappings.Find(mapping =>
+            mapping.Internal == managementPort && (requestPort == mapping.ExternalTlsProxy || requestPort == mapping.InternalTlsProxy));
+
+        if (_logger.IsEnabled(LogLevel.Trace) && portMapping != null)
+        {
+            _logger.LogTrace(
+                "Request received on port {RequestPort}. Allowed by CF_INSTANCE_PORTS mapping: [ Internal: {InternalPort}, ExternalTlsProxy: {ExternalTlsProxy}, InternalTlsProxy: {InternalTlsProxy} ]",
```

Comment:
I think we should either log `PortMapping.External` here as well, or leave out the property entirely.

Reply:
We don't currently have a use case for `External`